src/include/storage/store/chunked_group_undo_iterator.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <functional>
+
+#include "common/types/types.h"
+
+namespace kuzu {
+
+namespace transaction {
+class Transaction;
+}
+
+namespace storage {
+class ChunkedNodeGroup;
+class NodeGroupCollection;
+class ChunkedGroupUndoIterator;
+
+using chunked_group_undo_op_t = void (
+    ChunkedNodeGroup::*)(common::row_idx_t, common::row_idx_t, common::transaction_t);
+
+using chunked_group_iterator_construct_t = std::function<std::unique_ptr<ChunkedGroupUndoIterator>(
+    common::row_idx_t, common::row_idx_t, common::node_group_idx_t, common::transaction_t)>;
+
+// Note: these iterators are not necessarily thread-safe when used on their own
+class ChunkedGroupUndoIterator {
```

Comment:
I feel the class name `ChunkedGroupUndoIterator` a bit confusing to me.
1) It doesn't always refer to chunked group, instead it can be node group, or node table.
2) I don't catch the iteration part and the interface doesn't look like an iterator to me.

Should we rename it to a class name more along the line with `InsertionRollbackHandler`?